cloud/services/roleassignments/roleassignments_test.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package roleassignments
+
+import (
+	"context"
+	"github.com/Azure/azure-sdk-for-go/profiles/2019-03-01/authorization/mgmt/authorization"
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-06-01/compute"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/gomega"
+	"k8s.io/klog/klogr"
+	"net/http"
+	azure "sigs.k8s.io/cluster-api-provider-azure/cloud"
+	"testing"
+
+	"sigs.k8s.io/cluster-api-provider-azure/cloud/services/roleassignments/mock_roleassignments"
+	"sigs.k8s.io/cluster-api-provider-azure/cloud/services/virtualmachines/mock_virtualmachines"
+)
+
+func TestReconcileVM(t *testing.T) {
+	testcases := []struct {
+		name          string
+		expect        func(s *mock_roleassignments.MockRoleAssignmentScopeMockRecorder, m *mock_roleassignments.MockClientMockRecorder, v *mock_virtualmachines.MockClientMockRecorder)
+		expectedError string
+	}{
+		//{
+		//	name:          "role assignment already exists",
```

Comment:
don't think that's needed

Reply:
oops forgot to remove... I was wondering, what's the Create() behavior if an identity already exists on that VM? Does it return an error? Not really handling that case right now

Reply:
I think part of the problem is that we use a unique UUID every time we do the create so it's not deterministic, if we run the reconcile again it will create a new role assignment with a different UUID. I might change it so that the UUID is part of the spec so it only creates one.